tests/test_geodesic/test_geodesic.py

```diff
@@ -169,3 +170,34 @@ def test_kerr0_eq_kn00():
 
     assert_allclose(k.trajectory[0], kn.trajectory[0], atol=1e-6, rtol=1e-6)
     assert_allclose(k.trajectory[1], kn.trajectory[1], atol=1e-6, rtol=1e-6)
+
+
+def test_kerr0_eq_custom():
+    metric_params = (0.,)
+    q0 = [4., np.pi / 2, 0.]
+    p0 = [0., 0., 0.]
+
+    k = Timelike(
```

Comment:
I don't think, we need to check for `c` being same as `k`, since this is essentially an identity (because under the hood, `"Kerr"` is using `_kerr`), unlike say, `"Kerr"` with `a = 0` being the same as `"Schwarzschild"` etc. All we need to test are the following:

1. Whether `Timelike` initializes without errors with `metric = _kerr`
2. `c.metric_name == _kerr.__name__`
3. `c.metric == _kerr`

Something like the following test should be sufficient. This also saves a tiny amount of time compared to computing the geodesic twice.

```py
def test_custom_callable_metric():
    metric_params = (0.,)
    q0 = [4., np.pi / 2, 0.]
    p0 = [0., 0., 0.]

    try:
        c = Timelike(
            metric=_kerr,
            metric_params=metric_params,
            position=q0,
            momentum=p0,
            steps=50,
            delta=0.5,
            return_cartesian=True,
            suppress_warnings=True,
        )
        assert c.metric_name == _kerr.__name__
        assert c.metric == _kerr

    except:
        assert False
```
There are better ways to handle exceptions with `assert` but for now, something like this would be okay.